noCircularImportsRule.ts

```diff
@@ -113,6 +136,26 @@ function walk(context: Lint.WalkContext<Options>) {
 
     addToGraph(fileName, resolvedImportFileName, node)
   }
+
+  function getAllCycles(moduleName: string, accumulator: string[] = [], iterationDepth = 0): string[][] {
+    const moduleImport = imports.get(moduleName)
+    if (!moduleImport) return []
+    if (accumulator.indexOf(moduleName) !== -1)
+      return [accumulator]
+
+    if (iterationDepth >= context.options.searchDepthLimit)
+      return []
```

Comment:
Worth a `console.warn`?

Reply:
Ideally no. Large codebases will hit this a lot.

Reply:
With this change, when I have a cycle in a big codebase, will I ever find out about it?

Reply:
Or do you think the log wouldn't be useful, since there likely isn't a cycle (just deep dep graph?

Reply:
Yeah that's what I'm thinking. From working in code bases with a few hundred or a few thousand files, most of them just import a few around their own directory. Even `50` is pretty extreme.

...then again... you're right that if this results in nothing, but we know there was at least one cycle and it happens to be ridiculously long, I don't know what we could do intelligently show that.

Sorry for the brevity, on mobile :)